en/micro-integrator/docs/references/connectors/amazons3-connector/amazons3-connector-example.md

```diff
@@ -0,0 +1,308 @@
+# Amazon S3 Connector Example 
+
+The AmazonS3 Connector allows you to access the REST API of Amazon Simple Storage Service (Amazon S3), which provides a simple web services interface that can be used to store and retrieve any amount of data, at any time, from anywhere on the web. It gives any developer access to the same highly scalable, reliable, secure, fast, inexpensive infrastructure that Amazon S3 uses to run its own global network of websites. The service aims to maximize benefits of scale and to pass on those benefits to developers.
+
+Hence WSO2 EI AmazonS3 Connector is useful to take your on-premise data to the cloud. The advantage is, you do not need to worry about managing and replicating data on-premise. 
+
+## What you'll build
+
+This example depicts you how to use AmazonS3 connector to 
+
+1. Create a S3 bucket (a location for storing your data) in Amazon cloud 
+2. Upload a message into the created bucket as a text file
+3. Retrieve created text file back and convert into a message in WSO2 EI integration. 
+
+All three operations are exposed via an API. The API with the context `/s3connector` has three resources  
+
+* `/createbucket` - Once invoked, it will create a bucket in Amazon with the specified name
+* `/addobject`  - The incoming message will be stored into the specified bucket with the specified name
+* `/file/{filename}` - Once invoked, it will read the specified file from the specified bucket and respond with the content of the file
+
+<img src="/assets/img/connectors/Amazon-s3-diagram.png" title="Overview of Amazon S3 use case" width="800" alt="Amazon S3 use case"/>
```

Comment:
We need to explain this a bit more. It's not clear how all those three actions are represented by that first arrow and we need to go through the flow and explain a bit. Some improvements to the diagram might also be needed to enhance the clarity here.

Reply:
Updated.